doc/src/ch02-01-generic.md

```diff
@@ -1 +1,23 @@
 # Monitoring a file descriptor
+The `Generic` event source wraps a file descriptor ("fd") and fires its callback any time there are events on it ie. becoming readable or writable, or encountering an error. It's pretty simple, but it's what every other event source is based around. And since the platforms that calloop runs on expose many different kinds of events via fds, it's usually the key to using those events in calloop.
+
+For example on Linux, fd-based interfaces are available for GPIO, I2C, USB, UART, network interfaces, timers and many other systems. Integrating these into calloop starts with obtaining the appropriate fd, creating a `Generic` event source from it, and building up a more useful, abstracted event source around that. A detailed example of this is [given for ZeroMQ](ch04-00-a-full-example-zeromq.md). 
+
+## Creating a Generic source
+
+Creating a `Generic` event source requires three things:
+- The fd itself, of course
+- The ["interest"](api/calloop/struct.Interest.html) you have in this descriptor — this means, whether you want to generate events when the fd becomes readable, writeable, or both
+- The ["mode"](api/calloop/enum.Mode.html) of event generation - level triggered or edge triggered. If you are familiar with hard descriptor languages, you can think of this as the sensitivity.
```

Comment:
> If you are familiar with hard descriptor languages, you can think of this as the sensitivity.

I am not and that means nothing to me. Could we maybe instead make the explanation more explicit?

Reply:
Well, I meant to write "hardWARE descriptor langauges such as VHDL", but it's not really necessary. In retrospect I don't think it adds much.